transforms/universal/pdf2md/python/src/pdf2md_transform.py

```diff
@@ -0,0 +1,347 @@
+
+                except Exception as e:
+                    failed_doc_id.append(archive_doc_filename)
+                    logger.warning(
+                        f"Exception {str(e)} processing file {archive_doc_filename}, skipping"
+                    )
+
+
+            # Process ZIP archive of PDF documents
+            elif root_kind is not None and root_kind.mime == "application/zip":
+                logger.debug(
+                    f"Detected root file {file_name=} as ZIP. Iterating through the archive content."
+                )
+
+                with zipfile.ZipFile(io.BytesIO(byte_array)) as opened_zip:
+                    zip_namelist = opened_zip.namelist()
+
+                    for archive_doc_filename in zip_namelist:
+
+                        logger.info(
+                            "Processing "
+                            f"{archive_doc_filename=} "
+                        )
+
+                        with opened_zip.open(archive_doc_filename) as file:
+                            try:
+                                # Read the content of the file
+                                content_bytes = file.read()
+
+                                # Detect file type
+                                kind = filetype.guess(content_bytes)
+                                if kind is None or kind.mime != "application/pdf":
+                                    logger.info(
+                                        f"File {archive_doc_filename=} is not detected as PDF but {kind=}. Skipping."
+                                    )
+                                    skipped_doc_id.append(archive_doc_filename)
+                                    continue
+
+                                ext = kind.extension
+
+                                file_data = self._convert_pdf2md(doc_filename=archive_doc_filename, ext=ext, content_bytes=content_bytes)
+                                file_data["source_filename"] = TransformUtils.get_file_basename(file_name)
+
+                                data.append(file_data)
+                                number_of_rows += 1
+
+                            except Exception as e:
+                                failed_doc_id.append(archive_doc_filename)
+                                logger.warning(
+                                    f"Exception {str(e)} processing file {archive_doc_filename}, skipping"
+                                )
+            
+            else:
+                logger.warning(
+                    f"File {file_name=} is not detected as PDF nor as ZIP but {kind=}. Skipping."
+                )
+
+            table = pa.Table.from_pylist(data)
+            metadata = {
+                "nrows": len(table),
+                "nsuccess": len(success_doc_id),
+                "nfail": len(failed_doc_id),
+                "nskip": len(skipped_doc_id),
+            }
+            return [
+                (TransformUtils.convert_arrow_to_binary(table=table), ".parquet")
```

Comment:
This transform produces .parquet, not .md.  As such it should be renamed to pdf2parquet, similar to code2parquet.